lib/rbUtil.js

```diff
@@ -108,31 +108,24 @@ rbUtil.tidFromDate = function tidFromDate(date) {
         throw new Error('Invalid date');
     }
     // Create a new, deterministic timestamp
-    return uuid.v1({
-        node: [0x01, 0x23, 0x45, 0x67, 0x89, 0xab],
-        clockseq: 0x1234,
-        msecs: +date,
-        nsecs: 0
-    });
+    return uuid.TimeUuid.fromDate(date,
+        0,
+        new Buffer([0x01, 0x23, 0x45, 0x67, 0x89, 0xab]),
+        new Buffer([0x12, 0x34])).toString();
 };
 
-var uuidRe = /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/;
 /**
  * Check if a string is a valid timeuuid
  */
 rbUtil.isTimeUUID = function (s) {
-    return uuidRe.test(s);
+    return uuid.TimeUuid.test(s);
 };
 
-
-var reqIdBuf = new Buffer(16);
-
 /**
  * Generates a new request ID
  */
 rbUtil.generateRequestId = function() {
-    uuid.v4(null, reqIdBuf);
-    return reqIdBuf.toString('hex');
+    return uuid.Uuid.random().toRawString();
```

Comment:
Actually.. should we switch to a v1 uuid as well? It could be useful to have the start time encoded in the request ID.